src/rc_compat.c

```diff
@@ -88,32 +89,78 @@ struct tm* rc_gmtime_s(struct tm* buf, const time_t* timer)
 
 #if defined(_WIN32)
 
-/* https://gist.github.com/roxlu/1c1af99f92bafff9d8d9 */
+/* https://learn.microsoft.com/en-us/archive/msdn-magazine/2012/november/windows-with-c-the-evolution-of-synchronization-in-windows-and-c */
+/* implementation largely taken from https://github.com/libsdl-org/SDL/blob/0fc3574/src/thread/windows/SDL_sysmutex.c */
+
+#if defined(WINVER) && WINVER >= 0x0600
+
+void rc_mutex_init(rc_mutex_t* mutex)
+{
+  InitializeSRWLock(&mutex->srw_lock);
+  /* https://learn.microsoft.com/en-us/windows/win32/procthread/thread-handles-and-identifiers */
+  /* thread ids are never 0 */
+  mutex->owner = 0;
+  mutex->count = 0;
+}
+
+void rc_mutex_destroy(rc_mutex_t* mutex)
+{
+  /* Nothing to do here */
+  (void)mutex;
+}
+
+void rc_mutex_lock(rc_mutex_t* mutex)
+{
+  DWORD current_thread = GetCurrentThreadId();
+  if (mutex->owner == current_thread) {
+    ++mutex->count;
+    assert(mutex->count > 0);
```

Comment:
Missing include.
```
1>E:\Source\RetroAchievements\RAIntegration\rcheevos\src\rc_compat.c(116,5): warning C4013: 'assert' undefined; assuming extern returning int
```

Reply:
That's odd, there's an `#include <assert.h>` at the top of the file, which should include assert just fine?

Reply:
Yep. I might have screwed something up copying the changes across.